### ISSUE TITLE
New contracts for shared bridge and hyperscaling

### ISSUE THREAD

**Issue by kelemeno** [zksync-sdk/zksync2-js#19]
### 🌟 Feature Request 

We need the shared bridge contracts on L1 for hyperscaling. Messages, deposits and withdrawals will go to all hyperchains via a shared Bridgehub contract, and the assets will be locked in shared bridges. This means the contracts change, so this repo also has to change.  

#### 📝 Description

The main difference is we will have to specify the chainId of the hyperchain we are depositing to, this is in [src/adapters.ts](https://github.com/zksync-sdk/zksync2-js/blob/main/src/adapters.ts) . For backwards compatibility I think we can preserve the old methods and point them to Era.

**Comment by danijelTxFusion** [comment]
@kelemeno Are mentioning about [this](https://github.com/matter-labs/era-contracts/pull/29/files#diff-65c9adad72397ad27ac44d55c571728bbb033662e52e4cc06566eacc1e6ba746R197) change.

**Comment by kelemeno** [comment]
Yes, I have a local version but eventually it will need to be updated here as well.
